doc/rtd/explanation/format.rst

```diff
@@ -159,14 +159,60 @@ a MIME archive.
 ``cloud-boothook``
 ==================
 
-This content is `boothook` data. It is stored in a file under
-:file:`/var/lib/cloud` and executed immediately. This is the earliest `hook`
-available. Note, that there is no mechanism provided for running only once. The
+This content is `boothook` data.
+This is a bash script that runs during every boot.
+This is the earliest `hook` to run, it runs during the `Network boot stage`.
+
+You can use this to configure some network configuration during boot.
+
+Example
+-------
+
+.. code-block:: bash
+
+   #cloud-boothook
+   #!/bin/sh
+
+   sudo ufw enable
+   sudo ufw logging on
+
+   sudo ufw allow http
+   sudo ufw allow "OpenSSH"
+
+   sudo iptables -I DOCKER-USER -j ACCEPT
+
+Note, there is no mechanism provided for running only once. The
 `boothook` must take care of this itself.
 
 It is provided with the instance id in the environment variable
 ``INSTANCE_ID``. This could be made use of to provide a 'once-per-instance'
-type of functionality.
+type of functionality. An example of a `once-per-instance` script:
+
+Example
+-------
+
+.. code-block:: bash
+
+   #cloud-boothook
+   #!/bin/sh
+
+   _id=""
+   if [ -r /var/lib/my-instance-id ]
+     then
+       _id=$(cat /var/lib/my-instance-id)
+   fi
+
+   if [ -z $_id ]  || [ $INSTANCE_ID != $_id ]
+       then
+          sudo ufw enable
+          sudo ufw logging on
+
+          sudo ufw allow http
+          sudo ufw allow "OpenSSH"
+
+          sudo iptables -I DOCKER-USER -j ACCEPT
+   fi
+   sudo echo $INSTANCE_ID > /var/lib/my-instance-id
 
 Begins with: ``#cloud-boothook`` or ``Content-Type: text/cloud-boothook`` when
```

Comment:
I think that including the MIME information here adds more questions than answers. I know that other sections follow this pattern as well, but I really don't think that any of them should do this. MIME is, by definition, aggregation of different file types. The MIME section should describe how to aggregate different types rather than spread its details across other user data formats.

This information is only relevant to MIME users, so I don't think that we should include it here.

@s-makin Thoughts on this?

Reply:
I agree there's a lot of scope for improving the general layout and organisation of this page, and I'm happy for you guys to make any improvements as you see fit while you're reworking it anyway.

If you feel it's beyond the scope of this PR that's also fine - we'll be overhauling all these pages over the next few weeks so if it's not done now it'll be done then. I'll tag you both in that effort anyway :)

Reply:
I have updated the MIME archive documentation as well.